lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php

```diff
@@ -66,22 +74,38 @@ public function getValue($object = null)
     public function setValue($object, $value = null): void
     {
         if ($value !== null) {
-            $enumType = $this->enumType;
-            try {
-                $value = $enumType::from($value);
-            } catch (ValueError $e) {
-                assert(is_string($value) || is_int($value));
-
-                throw MappingException::invalidEnumValue(
-                    get_class($object),
-                    $this->originalReflectionProperty->getName(),
-                    (string) $value,
-                    $enumType,
-                    $e
-                );
+            if (is_array($value)) {
+                $value = array_map(function ($item) use ($object): mixed {
```

Comment:
```suggestion
                $value = array_map(function ($item) use ($object): BackedEnum {
```

Reply:
Changed in [b6e7472](https://github.com/doctrine/orm/pull/9497/commits/b6e7472f91ffe29851eb08093f6f8c1e8eb817a8)